packages/cli/src/cli.ts

```diff
@@ -21,68 +21,215 @@ import DefaultServient from "./cli-default-servient";
-                try {
-                    config = JSON.parse(data);
-                    if (typeof config !== "object" || Array.isArray(config)) {
-                        throw new Error("Invalid configuration file");
-                    }
-                } catch (err) {
-                    reject(err);
-                }
-                console.info("[cli]", `WoT-Servient using config file '${open}'`);
-                resolve(config);
-            }
-        });
-    });
+        const data = fs.readFileSync(open, "utf-8");
+        if (!schemaValidator(JSON.parse(data))) {
+            throw new InvalidArgumentError(
+                `Config file contains invalid an JSON: ${(schemaValidator.errors ?? [])
+                    .map((o: ErrorObject) => o.message)
+                    .join("\n")}`
+            );
+        }
+        return filename;
+    } catch (err) {
+        throw new InvalidArgumentError(`Error reading config file: ${err}`);
+    }
+}
+function parseConfigParams(param: string, previous: any) {
+    // Validate key-value pair
+    if (!/^([a-zA-Z0-9_.]+):=([a-zA-Z0-9_]+)$/.test(param)) {
+        throw new InvalidArgumentError("Invalid key-value pair");
+    }
+    const fieldNamePath = param.split(":=")[0];
+    const fieldNameValue = Number(param.split(":=")[1]) ? +param.split(":=")[1] : param.split(":=")[1];
+
+    // Build object using dot-notation JSON path
+    const obj = _.set({}, fieldNamePath, fieldNameValue);
+    if (!schemaValidator(obj)) {
+        throw new InvalidArgumentError(
+            `Config parameter '${param}' is not valid: ${(schemaValidator.errors ?? [])
+                .map((o: ErrorObject) => o.message)
+                .join("\n")}`
+        );
+    }
+    // Concatenate validated paramters
+    let result = previous ?? {};
+    result = _.merge(result, obj);
+    return result;
 }
 
-function overrideConfig(conf: Record<string, unknown>) {
-    conf = conf ?? {};
-
-    servientPorts.forEach((port, protocol) => {
-        if (port !== undefined) {
-            if (!(protocol in conf)) {
-                conf[protocol] = {};
-            }
-            (conf[protocol] as { port: number }).port = port;
+// CLI options declaration
+program
+    .option("-i, --inspect [host]:[port]", "activate inspector on host:port (default: 127.0.0.1:9229)", parseIp)
+    .option("-ib, --inspect-brk [host]:[port]", "activate inspector on host:port (default: 127.0.0.1:9229)", parseIp)
+    .option("-c, --client-only", "do not start any servers (enables multiple instances without port conflicts)")
+    .option("-cp, --compiler <module>", "load module as a compiler")
+    .option(
+        "-f, --config-file <file>",
+        "load configuration from specified file",
+        parseConfigFile,
+        "wot-servient.conf.json"
+    )
+    .option(
+        "-p, --config-params <param...>",
+        "override configuration paramters [key1:=value1 key2:=value2 ...] (e.g. http.port=8080)",
```

Comment:
@relu91 can you also update this line? because I previously used this notation `key1:=value1` but then the example right after is `http.port=8080` instead of `http.port:=8080`